server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -187,7 +187,7 @@ private RemoteRestoreResult executeRestore(
             IndexMetadata indexMetadata = indexMetadataEntry.getValue().v2();
             boolean metadataFromRemoteStore = indexMetadataEntry.getValue().v1();
             IndexMetadata updatedIndexMetadata = indexMetadata;
-            if (restoreAllShards || metadataFromRemoteStore) {
+            if (restoreAllShards) {
                 updatedIndexMetadata = IndexMetadata.builder(indexMetadata)
                     .state(IndexMetadata.State.OPEN)
                     .version(1 + indexMetadata.getVersion())
```

Comment:
it was decided not to bump up the versions, you are addressing it somewhere else?

Reply:
we removed the metadataFromRemoteStore flag in the if block above. whenever remote cluster state restore is done metadataFromRemoteStore is true. so now the versions wont be bumped.

Reply:
Thats a sneaky check, in case of metadataFromRemoteStore is true and restoreAllShards is true, it will bump the version which is not very explicit.

Reply:
okay. let me update to `metadataFromRemoteStore == false && restoreAllShards`